mad-dog/src/tests/test_integration.js

```diff
@@ -350,10 +363,13 @@ module.exports = coreIntegration = async ({
       requesterIdx,
       walletIdMap[requesterIdx]
     )
-    emit(Event.REQUEST, trackUploadRequest)
-    emit(Event.REQUEST, trackRepostRequest)
-    emit(Event.REQUEST, createPlaylistRequest)
-    emit(Event.REQUEST, addPlaylistTrackRequest)
+    const requests = [trackUploadRequest, trackRepostRequest, createPlaylistRequest, addPlaylistTrackRequest]
+    for (const request of requests) {
+      const randomEmit = _.random(1)
```

Comment:
should there be some conditional set here? as in - there can't be a trackRepost request until tracks have been uploaded etc. or maybe i have misunderstood

Reply:
Here it's just randomly emitting requests and the request listener actually determines how to handle each request. I think it would be complicated to pre-determine that for all request types. This might simulate live traffic more as well.

Reply:
good point, i totally agree